pkg/common-controller/snapshot_controller.go

```diff
@@ -1421,7 +1483,12 @@ func (ctrl *csiSnapshotCommonController) checkAndSetInvalidContentLabel(content
 		}
 		contentClone.ObjectMeta.Labels[utils.VolumeSnapshotContentInvalidLabel] = ""
 	}
-	updatedContent, err := ctrl.clientset.SnapshotV1beta1().VolumeSnapshotContents().Update(context.TODO(), contentClone, metav1.UpdateOptions{})
+	var newcontent *crdv1.VolumeSnapshotContent
+	patch, err := createContentPatch(content, newcontent)
```

Comment:
I think this is wrong.  `newcontent` should not be nil.  It should be `contentClone` which has the updated information.
I see the same problem in other places as well.

Neither `content` nor `newcontent` should be nil.  Otherwise there is no need to merge and patch.

Reply:
@xing-yang just for confirmation, here the content clone should be the one's which user perform patch operation on volumesnapshot and volumesnapshotcontent ?

Reply:
The one with new data should be passed in as the 2nd parameter.  As shown on line 1484, contentClone is modified with a new label.  We need to have this label in the updated content.

So you'll need to look at each case and decide.

Reply:
See this original line `updatedContent, err := ctrl.clientset.SnapshotV1beta1().VolumeSnapshotContents().Update(context.TODO(), contentClone, metav1.UpdateOptions{})`, `contentClone` is used to do an update.  That is the indication that contentClone has the new data.

Reply:
Here's a good example to show what is the oldData and what is the newData.  I'd suggest that you start with one update and see if it works before testing all the changes.
https://github.com/kubernetes/kubernetes/blob/v1.20.0-beta.1/pkg/volume/util/resize_util.go#L57

Reply:
@xing-yang  sure , thanks for suggestions i will update  this accordingly

Reply:
@xing-yang  got it like wise in this case https://github.com/kubernetes/kubernetes/blob/v1.20.0-beta.1/pkg/volume/util/resize_util.go#L80 `patchbytes` is refer as resouces is  patched!

Reply:
i have written down the functionalities as mention in line https://github.com/kubernetes/kubernetes/blob/v1.20.0-beta.1/pkg/volume/util/resize_util.go#L75 for volumesnapshot and volumesnapshot content, but i got as null reference , i will be looking into this

Thanks